src/pytest_qgis/pytest_qgis.py

```diff
@@ -317,6 +317,11 @@ def _show_qgis_dlg(common_settings: Settings, qgis_parent: QWidget) -> None:
     if not common_settings.qgis_init_disabled:
         qgis_parent.setWindowTitle("Test QGIS dialog opened by Pytest-qgis")
         qgis_parent.show()
+
+        # Process events each time layer a visible layer is added to
+        # be able to change the extent properly
+        assert _APP
+        QgsProject.instance().legendLayersAdded.connect(_APP.processEvents)
```

Comment:
Ah, you are right, my mistake! I originally put this in `_configure_qgis_map` but since it is only needed when the `qgis_show_map` marker is used (which is not that often I think) I moved it away. But I'll move it back there.